### PR TITLE
[DOC] Add model usage examples (v1)

### PR DESCRIPTION
**Pull request by ramanbansal1**
#### Reference Issues/PRs

 Related to:  #2377 

 #### What does this implement/fix? Explain your changes.

 This PR adds doctest-style usage examples for 4 v1 models as part of the ongoing effort to improve documentation
 and user onboarding. The examples are minimal, runnable snippets that demonstrate typical model instantiation,
 training, and prediction flows using the v1 API.

- Added usage examples for v1 models:
     - DeepAR (DeepAR_pkg)
     - DecoderMLP (DecoderMLP_pkg)
     - NBeats (NBeats_pkg)
     - NBeatsKAN (NBeatsKAN_pkg)

#### Each example contains:

   - Minimal synthetic dataset generation using  generate_ar_data 
   - TimeSeriesDataSet creation with appropriate v1 parameters
   - Model initialization using  from_dataset()  method
   - Lightning Trainer setup for training
   - Prediction workflow demonstration

#### The examples are doctest-compliant:

   - Use proper doctest prompts (>>> and ...) for doctest runner compatibility
   - Mark Lightning-dependent operations (fit/predict) with  # doctest: +SKIP  to avoid dependency issues in CI
   - All lines ≤ 88 characters to comply with repository linting rules
   - Use the standard v1 API patterns (TimeSeriesDataSet + from_dataset)

#### What should a reviewer concentrate their feedback on?

   - Examples correctness:
     - Are the imports and model calls accurate for the current v1 codebase?
     - Do the TimeSeriesDataSet parameters match each model's requirements?
     - Are the loss functions appropriate for each model type?
   - CI compliance:
     - Do the doctest examples parse correctly with  python -m doctest ?
     - Are the SKIP directives appropriately placed for Lightning-dependent operations?
     - Do line lengths comply with the 88-character limit?
   - API consistency:
     - Do the examples correctly use the v1  from_dataset()  pattern?
     - Are the dataset parameters (encoder/prediction lengths, features) correctly configured?
   - Documentation quality:
     - Are the examples minimal yet complete enough to be copy-paste runnable?
     - Do they provide sufficient context for users to understand the v1 workflow?

#### Did you add any tests for the change?

   - No. These are documentation docstring examples only. The existing test suite for v1 models will validate the
   underlying functionality, and the repository's doctest runner will validate the docstring examples.

#### Any other comments?

   - This is a focused documentation PR addressing 4 of the v1 models mentioned in issue #2377.
   - The examples use the same synthetic data generation pattern for consistency.
   - NBeatsKAN example excludes the GridUpdateCallback due to compatibility issues in the current implementation.
   - Future work could add examples for the remaining v1 models mentioned in the original issue (NHiTS, RNN, TFT,
   TiDE, TimeXer, xLSTMTime).

#### PR checklist

   - [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. ([DOC])
   - [ ] Added/modified tests
   - [ ] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with  p
   re-commit install . To run hooks independent of commit, execute  pre-commit run --all-files

### CODE REVIEW

File: pytorch_forecasting/models/deepar/_deepar_pkg.py [hunk 1]
```diff
@@ -4,7 +4,75 @@
...
+    >>> model = DeepAR.from_dataset(
+    ...     training,
+    ...     learning_rate=1e-2,
+    ...     hidden_size=16,
+    ...     rnn_layers=2,
+    ...     loss=MultivariateNormalDistributionLoss(rank=10),
+    ...     optimizer="Adam",
+    ... )
+    >>>
+    >>> # Training with Lightning - skip in doctest
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/deepar/_deepar_pkg.py]: please dont add "skip in doctest" etc comments in the examples as they are displayed in the documentation and are not needed there. THe way you added `# doctest: +SKIP` is the best way to skip the doc tests

File: pytorch_forecasting/models/deepar/_deepar_pkg.py [hunk 2]
```diff
@@ -4,7 +4,75 @@
...
+    >>> # import lightning as L  # doctest: +SKIP
+    >>> # trainer = L.Trainer(max_epochs=1, accelerator="cpu")  # doctest: +SKIP
+    >>> # trainer.fit(model, train_dataloaders=train_dataloader,  # doctest: +SKIP
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/deepar/_deepar_pkg.py]: Please dont add commented out code in the example. Also why should the doc test be skipped here?

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
I think these examples should go in model classes instead of `pkg` classes?
Also, please dont add commented out  code in the examples

File: pytorch_forecasting/models/deepar/_deepar_pkg.py (hunk 2 above)
Comment (ramanbansal1) [reply pytorch_forecasting/models/deepar/_deepar_pkg.py]: okky i have removed that

File: pytorch_forecasting/models/deepar/_deepar_pkg.py (hunk 1 above)
Comment (ramanbansal1) [reply pytorch_forecasting/models/deepar/_deepar_pkg.py]: understood

File: pytorch_forecasting/models/deepar/_deepar_pkg.py (hunk 2 above)
Comment (ramanbansal1) [reply pytorch_forecasting/models/deepar/_deepar_pkg.py]: When i dont keep it i get ci/cd failures

**Review by phoeenniixx — APPROVED** [review]
Thanks!
Just changed the positions of the example blocks in `nbeats` and `nbeats_kan`
